### ISSUE TITLE
MongoDB high availability software RAID creation failure

### ISSUE THREAD

**Issue by iwuvjhdva** [Azure/azure-quickstart-templates#647]
Hi, I get the following error after trying to install the template, could you please assist.

2015/09/24 12:11:38 Configuring a replica set rs0
2015/09/24 12:11:38 Stopping MongoDB daemon processes (PID 2353)
2015/09/24 12:11:38 Starting MongoDB daemon processes
2015/09/24 12:11:38 mongod start/running, process 2454
2015/09/24 12:11:38
2015/09/24 12:11:38 ---errout---
2015/09/24 12:11:38 E: Failed to fetch http://azure.archive.ubuntu.com/ubuntu/pool/main/m/mdadm/mdadm_3.2.5-5ubuntu4.1_amd64.deb  404  Not Found
2015/09/24 12:11:38
2015/09/24 12:11:38 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
2015/09/24 12:11:38 Device contains neither a valid DOS partition table, nor Sun, SGI or OSF disklabel
2015/09/24 12:11:38 Building a new DOS disklabel with disk identifier 0xba79b2bf.
2015/09/24 12:11:38 Changes will remain in memory only, until you decide to write them.
2015/09/24 12:11:38 After that, of course, the previous content won't be recoverable.
2015/09/24 12:11:38
2015/09/24 12:11:38 Warning: invalid flag 0x0000 of partition table 4 will be corrected by w(rite)
2015/09/24 12:11:38 Device contains neither a valid DOS partition table, nor Sun, SGI or OSF disklabel
2015/09/24 12:11:38 Building a new DOS disklabel with disk identifier 0xef560263.
2015/09/24 12:11:38 Changes will remain in memory only, until you decide to write them.
2015/09/24 12:11:38 After that, of course, the previous content won't be recoverable.
2015/09/24 12:11:38
2015/09/24 12:11:38 Warning: invalid flag 0x0000 of partition table 4 will be corrected by w(rite)
2015/09/24 12:11:38 ./vm-disk-utils-0.1.sh: line 280: mdadm: command not found
2015/09/24 12:11:38 mke2fs 1.42.9 (4-Feb-2014)
2015/09/24 12:11:38 Could not stat /dev/md0 --- No such file or directory
2015/09/24 12:11:38
2015/09/24 12:11:38 The device apparently does not exist; did you specify it correctly?

**Comment by singhkay** [comment]
@iwuvjhdva There's an internal thread about this with the same error. Here's what the template author @TheAzureGuy suggested 

Our MondoDB template is operating normally, and we don’t see any issues with the implementation of the template. However, what’s happening here is that the disk partitioning script (vm-disk-utils-0.1.sh) was unable to download the mdadm package. Since data disk was not correctly formatted and prepared, mongodb installation failed. It also led to some other abnormalities like not being able to setup the admin user.

In summary, we need to resolve this error first, and it should automatically cure the rest.

I suspect the following command has failed execution:

    apt-get -y install mdadm

Please can you ask your customer to run the above command from within an SSH session and send me its output. Then, please ask him to invoke this command and let me know what it comes back with.

    apt-get -y install mdadm --fix-missing

**Comment by trentmswanson** [comment]
I could not reproduce this issue.  The requested changes to the partitioning/formatting script are part of PR #675 submitted a couple days ago.
